batch2/batch/front_end/front_end.py

```diff
@@ -581,59 +592,38 @@
     if not record:
         raise web.HTTPNotFound()
 
-    state = record['state']
-    ip_address = record['ip_address']
+    running_status = await _get_job_running_status(record)
+    return job_record_to_dict(record, running_status)
 
-    status = record['status']
-    if status is not None:
-        status = json.loads(status)
 
-    if state == 'Running':
-        assert status is None
-        async with aiohttp.ClientSession(
-                raise_for_status=True, timeout=aiohttp.ClientTimeout(total=60)) as session:
-            try:
-                url = (f'http://{ip_address}:5000'
-                       f'/api/v1alpha/batches/{batch_id}/jobs/{job_id}/status')
-                resp = await request_retry_transient_errors(session, 'GET', url)
-                status = await resp.json()
-            except aiohttp.ClientResponseError as e:
-                if e.status == 404:
-                    status = None
-                else:
-                    raise
-
-    return status
-
-
-@routes.get('/api/v1alpha/batches/{batch_id}/jobs/{job_id}/status')
-@prom_async_time(REQUEST_TIME_GET_JOB_STATUS)
+@routes.get('/api/v1alpha/batches/{batch_id}/jobs/{job_id}')
+@prom_async_time(REQUEST_TIME_GET_JOB)
 @rest_authenticated_users_only
-async def get_job_status(request, userdata):
+async def get_job(request, userdata):
     batch_id = int(request.match_info['batch_id'])
     job_id = int(request.match_info['job_id'])
     user = userdata['username']
-    status = await _get_job_status(request.app, batch_id, job_id, user)
+
+    status = await _get_job(request.app, batch_id, job_id, user)
     return web.json_response(status)
 
 
-@routes.get('/batches/{batch_id}/jobs/{job_id}/status')
-@prom_async_time(REQUEST_TIME_GET_JOB_STATUS_UI)
-@aiohttp_jinja2.template('job_status.html')
+@routes.get('/batches/{batch_id}/jobs/{job_id}')
+@prom_async_time(REQUEST_TIME_GET_JOB_UI)
 @web_authenticated_users_only()
-async def ui_get_job_status(request, userdata):
+async def ui_get_job(request, userdata):
     batch_id = int(request.match_info['batch_id'])
     job_id = int(request.match_info['job_id'])
     user = userdata['username']
 
-    status = await _get_job_status(request.app, batch_id, job_id, user)
-
+    job_status = await _get_job(request.app, batch_id, job_id, user)
     page_context = {
         'batch_id': batch_id,
         'job_id': job_id,
-        'job_status': json.dumps(status, indent=2)
+        'job_log': await _get_job_log(request.app, batch_id, job_id, user),
+        'job_status': json.dumps(job_status, indent=2)
     }
-    return await render_template('batch2', request, userdata, 'job_status.html', page_context)
+    return await render_template('batch2', request, userdata, 'job.html', page_context)
```

Comment:
I can't make the comment in the right place... I think you need to make the same UI changes you did for CI to batch2 where the `batch.html` page needs a different link and a combined log, status page (`job.html`). It looks like you have the reference to `job.html` already in the right place here.